admin/services/pin-generation.service.js

```diff
@@ -90,30 +94,39 @@ pinGenerationService.generatePupilPins = async (pupilsList) => {
 pinGenerationService.generateSchoolPassword = (school) => {
   let { schoolPin, pinExpiresAt } = school
   if (!pinValidator.isActivePin(schoolPin, pinExpiresAt)) {
-    const length = 8
-    school.schoolPin = pinGenerationService.generateRandomPin(length)
+    const allowedWords = (config.Data.allowedWords && config.Data.allowedWords.split(',')) || []
+    const firstRandomWord = allowedWords.length > 0
+      ? allowedWords[pinGenerationService.generateCryptoRandomNumber(0, allowedWords.length - 1)] : ''
+    const secondRandomWord = allowedWords.length > 0
+      ? allowedWords[pinGenerationService.generateCryptoRandomNumber(0, allowedWords.length - 1)] : ''
+    const numberCombination = randomGenerator.getRandom(2, chars)
+    school.schoolPin = `${firstRandomWord}${numberCombination}${secondRandomWord}`
     school.pinExpiresAt = fourPmToday()
   }
   return school
 }
 
 /**
- * Generate Pupil Pin
- * @returns {String}
- */
-pinGenerationService.generatePupilPin = () => {
-  const chars = '23456789'
-  return randomGenerator.getRandom(4, chars)
+ * Generating random numbers in specific range using crypto.randomBytes from crypto library
+ * Maximum available range is 281474976710655 or 256^6-1
+ * Maximum number for range must be equal or less than Number.MAX_SAFE_INTEGER (usually 9007199254740991)
+ * @param minimum
```

Comment:
The jsdoc needs to specify whether the max and min values in the range are inclusive or exclusive.    E.g. will they be in the output or not?